pkg/apis/networking/validation/validation_test.go

```diff
@@ -33,1022 +33,493 @@ import (
 	utilpointer "k8s.io/utils/pointer"
 )
 
+func makeValidNetworkPolicy() *networking.NetworkPolicy {
+	return &networking.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
+		Spec: networking.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"a": "b"},
+			},
+		},
+	}
+}
+
+func makeNetworkPolicyCustom(tweaks ...func(networkPolicy *networking.NetworkPolicy)) *networking.NetworkPolicy {
+	networkPolicy := makeValidNetworkPolicy()
+	for _, fn := range tweaks {
+		fn(networkPolicy)
+	}
+	return networkPolicy
+}
+
 func TestValidateNetworkPolicy(t *testing.T) {
 	protocolTCP := api.ProtocolTCP
 	protocolUDP := api.ProtocolUDP
 	protocolICMP := api.Protocol("ICMP")
 	protocolSCTP := api.ProtocolSCTP
 	endPort := int32(32768)
-	successCases := []networking.NetworkPolicy{
-		{
-			ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
-			Spec: networking.NetworkPolicySpec{
-				PodSelector: metav1.LabelSelector{
-					MatchLabels: map[string]string{"a": "b"},
-				},
-				Ingress: []networking.NetworkPolicyIngressRule{},
+
+	// Tweaks used below.
+	setIngressEmptyFirstElement := func(networkPolicy *networking.NetworkPolicy) {
+		networkPolicy.Spec.Ingress = []networking.NetworkPolicyIngressRule{{}}
+	}
+
+	setIngressEmptyFrom := func(networkPolicy *networking.NetworkPolicy) {
+		networkPolicy.Spec.Ingress[0].From = []networking.NetworkPolicyPeer{}
+	}
+
+	setIngressFromEmptyFirstElement := func(networkPolicy *networking.NetworkPolicy) {
+		networkPolicy.Spec.Ingress[0].From = []networking.NetworkPolicyPeer{{}}
+	}
+
+	setIngressEmptyPorts := func(networkPolicy *networking.NetworkPolicy) {
+		networkPolicy.Spec.Ingress[0].Ports = []networking.NetworkPolicyPort{}
+
+	}
+
+	setIngressPorts := func(networkPolicy *networking.NetworkPolicy) {
+		networkPolicy.Spec.Ingress[0].Ports = []networking.NetworkPolicyPort{
+			{
+				Protocol: nil,
+				Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 80},
```

Comment:
I think `intstr.FromInt(80)` should work?

Reply:
I am not that experienced with golang, and I was having trouble trying to do something like &intstr.fromInt(80), so I needed to declare them as variables on the top and do &port80, for example. I am not sure what is best here.